scripts/linters/general_purpose_linter.py

```diff
@@ -0,0 +1,730 @@
+        'excluded_files': ('python_utils.py', 'python_utils_test.py'),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'[^.|\w]input\('),
+        'message': 'Please use python_utils.INPUT.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'[^.|\w|\s]map\('),
+        'message': 'Please use python_utils.MAP.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'\Wnext\('),
+        'message': 'Please use python_utils.NEXT.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'object\):'),
+        'message': 'Please use python_utils.OBJECT.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'\Wrange\('),
+        'message': 'Please use python_utils.RANGE.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'\Wround\('),
+        'message': 'Please use python_utils.ROUND.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'\Wstr\('),
+        'message': (
+            'Please try to use python_utils.convert_to_bytes() for the strings '
+            'used in webapp2\'s built-in methods or for strings used directly '
+            'in NDB datastore models. If you need to cast ints/floats to '
+            'strings, please use python_utils.UNICODE() instead.'),
+        'excluded_files': ('python_utils.py'),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'\Wzip\('),
+        'message': 'Please use python_utils.ZIP.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'basestring'),
+        'message': 'Please use python_utils.BASESTRING.',
+        'excluded_files': ('python_utils.py'),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'__metaclass__'),
+        'message': 'Please use python_utils.with_metaclass().',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'iteritems'),
+        'message': 'Please use items() instead.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'itervalues'),
+        'message': 'Please use values() instead.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': re.compile(r'iterkeys'),
+        'message': 'Please use keys() instead.',
+        'excluded_files': (),
+        'excluded_dirs': ()
+    }
+]
+
+BAD_PATTERNS_MAP = {
+    '.js': BAD_PATTERNS_JS_AND_TS_REGEXP,
+    '.ts': BAD_PATTERNS_JS_AND_TS_REGEXP,
+    '.html': BAD_LINE_PATTERNS_HTML_REGEXP,
+    '.py': BAD_PATTERNS_PYTHON_REGEXP
+}
+
+
+def _is_filepath_excluded_for_bad_patterns_check(pattern, filepath):
```

Comment:
Let's make these public functions. Same for all of the ones below.

Reply:
Done.